ckan/cli/cli.py

```diff
@@ -72,6 +73,10 @@ def __init__(self, conf=None):
 @click_config_option
 @click.pass_context
 def ckan(ctx, config, *args, **kwargs):
+    # This is necessary to allow the user to create
+    # a config file when one isn't already present
+    if all(arg in sys.argv for arg in ['generate', 'config']):
```

Comment:
I think this idea is really nice, but there are two concerns from my side:

* Probably, the whole `generate` tree doesn't require config file, so i would only check if first sub-command is `generate`
* Also, there is a small possibility that some `generate` commands will either require config file, or behave differently when it provided. So i would add `config is None` check - if there is no config file AND it's a `generate` command - let's continue. But if there IS config file - let's use it- there is no harm in it

Reply:
@smotornyuk Sorry for the delay. I made the changes. Let me know what you think. Thanks!